src/hooks/useFeedbackDisclosure.ts

```diff
@@ -0,0 +1,75 @@
+import { useDisclosure } from "@chakra-ui/react"
+import { useLocation } from "react-router-dom"
+
+import { LOCAL_STORAGE_KEYS } from "constants/localStorage"
+
+import { useLoginContext } from "contexts/LoginContext"
+
+import { isEditPageUrl, isSpecialPagesUrl } from "utils/pages"
+
+import { useLocalStorage } from "./useLocalStorage"
+
+type LastSeenFeedbackTime = number
+type UserId = string
+
+type FeedbackStorageMappings = Record<UserId, LastSeenFeedbackTime>
+
+// 1 week in ms
+const NPS_SURVEY_DURATION = 7 * 24 * 60 * 60 * 1000
```

Comment:
changed for clarity cos i think the initial 3 weeks was abit confusing. as long as the last seen + duration is less than the current time, it means that they haven't done the survey in the current internval

Reply:
Based on the description,

> this is in intervals of 3 wks since the last seen timing

So this should be last seen + 3 weeks duration > current time right?

Reply:
changed PR desc, using 1 week now

Reply:
ok!